subprojects/core/src/main/java/org/gradle/internal/classpath/DefaultCachedClasspathTransformer.java

```diff
@@ -71,7 +71,7 @@ public DefaultCachedClasspathTransformer(
         this.globalCacheLocations = globalCacheLocations;
         this.cache = classpathTransformerCacheFactory.createCache(cacheRepository, fileAccessTimeJournal);
         this.fileAccessTracker = classpathTransformerCacheFactory.createFileAccessTracker(fileAccessTimeJournal);
-        this.executor = executorFactory.create("jar transforms");
+        this.executor = executorFactory.create("jar transforms", Runtime.getRuntime().availableProcessors());
```

Comment:
Shouldn't we be using worker leases here?

Reply:
I don't know. Can you make the case?

Reply:
My understanding is that we want all parallel work to be governed by worker leases. That way we can ensure that there's always at most `maxWorkers` number of parallel things happening at the same time. This code here seems no exception to that.

Reply:
@lptr, that's a good case. I've added #16181 so we can revisit it after we're done with all things 7.0.